Data/Aeson/Types/FromJSON.hs

```diff
@@ -1541,7 +1541,9 @@ instance FromJSONKey Float where
         _           -> Scientific.toRealFloat <$> parseScientificText t
 
 instance (FromJSON a, Integral a) => FromJSON (Ratio a) where
-    parseJSON (Number x) = pure (realToFrac x)
+    parseJSON (Number x) = prependContext "Ratio"
+                         $ withBoundedScientific' (pure . realToFrac)
```

Comment:
```
withBoundedScientific_ :: (Parser a -> Parser a) -> (Scientific -> Parser a) -> Value -> Parser a
withBoundedScientific_ whenFail f (Number scientific) =
    if exp10 > 1024
    then whenFail (fail msg)
    else f scientific
```

tests only for overflows, not undeerflows. I.e. `1e-2000` won't fail (but it should).

Reply:
You're right. Fixed this as well. Thank for pointing that stuff out